cinema/views.py

```diff
@@ -0,0 +1,38 @@
+from django.shortcuts import get_object_or_404
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework import status
+
+from cinema.models import Movie
+from cinema.serializers import MovieSerializer
+
+
+# Create your views here.
+@api_view(["GET", "POST"])
+def movie_list(request) -> Response:
+    if request.method == "GET":
+        movies = Movie.objects.all()
+        serializer = MovieSerializer(movies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    else:
+        serializer = MovieSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_detail(request, pk: int) -> Response:
+    movie = get_object_or_404(Movie, id=pk)
+    if request.method == "GET":
+        movie = get_object_or_404(Movie, id=pk)
```

Comment:
The `get_object_or_404` call is redundant here since `movie` is already retrieved at the beginning of the `movie_detail` function. You can remove this line to avoid unnecessary database queries.